src/hotspot/share/gc/shenandoah/shenandoahClosures.hpp

```diff
@@ -89,35 +89,30 @@ class ShenandoahUpdateRefsClosure: public ShenandoahOopClosureBase {
   inline void do_oop_work(T* p);
 };
 
-template <DecoratorSet MO = MO_UNORDERED>
-class ShenandoahEvacuateUpdateMetadataClosure: public ShenandoahOopClosureBase {
-private:
-  ShenandoahHeap* const _heap;
-  Thread* const         _thread;
+template <bool atomic>
+class ShenandoahEvacuateUpdateRootClosureBase : public ShenandoahOopClosureBase {
+protected:
+    ShenandoahHeap* const _heap;
```

Comment:
Here and later the indenting is a bit off: needs to be 2 spaces, not 4.

Reply:
Fixed